tedium.js

```diff
@@ -245,14 +342,23 @@ Promise.resolve().then(() => {
         })
     );
   }
-  return promiseAllWithProgress(promises, 'Cleaning...');
+  return promiseAllWithProgress(promises, 'Applying transforms...');
 }).then(() => {
-  if (elementsPushed === 0) {
+  console.log();
+  if (elementsPushed === 0 && pushesDenied === 0) {
     console.log('No changes needed!');
-  } else {
+  } else if (pushesDenied === 0) {
     console.log(`Successfully pushed to ${elementsPushed} repos.`);
+  } else if (opts.max_changes === 0) {
+    console.log(`${pushesDenied} changes ready to push. ` +
+                `Call with --max_changes=N to push them up!`);
+  } else {
+    console.log(`Successfully pushed to ${elementsPushed} repos. ` +
+                `${pushesDenied} remain.`);
```

Comment:
First: awesome script. Very efficient. I like that no time was spared for unnecessary / pre-mature design or optimization.

That said, all of the assorted global state that the _conclusion_ is derived from tickles my gag reflex a bit (on an instinctual level). I'm sure you've thought about potentially refactoring this a little. Just thought I would mention it.